read-fonts/src/tables/postscript/scale.rs

```diff
@@ -0,0 +1,541 @@
+//! Scaler for CFF outlines.
+
+use std::ops::Range;
+
+use super::{
+    dict::{self, Blues},
+    BlendState, Error, FdSelect, Index,
+};
+use crate::{
+    tables::{cff::Cff, cff2::Cff2, variations::ItemVariationStore},
+    types::{F2Dot14, Fixed, GlyphId, Pen},
+    FontData, FontRead, TableProvider,
+};
+
+/// Type for loading, scaling and hinting outlines in CFF/CFF2 tables.
+///
+/// # Subfonts
+///
+/// CFF tables can contain multiple logical "subfonts" which determine the
+/// state required for processing some subset of glyphs. This state is
+/// accessed using the [`FDArray and FDSelect`](https://adobe-type-tools.github.io/font-tech-notes/pdfs/5176.CFF.pdf#page=28)
+/// operators to select an appropriate subfont for any given glyph identifier.
+/// This process is exposed on this type with the
+/// [`subfont_index`](Self::subfont_index) method to retrieve the subfont
+/// index for the requested glyph followed by using the
+/// [`subfont`](Self::subfont) method to create an appropriately configured
+/// subfont for that glyph.
+///
+/// # Example
+///
+/// ```
+/// # use read_fonts::{tables::postscript::*, types::*, FontRef};
+/// # fn example(font: FontRef, coords: &[F2Dot14], pen: &mut impl Pen) -> Result<(), Error> {
+/// let scaler = Scaler::new(&font)?;
+/// let glyph_id = GlyphId::new(24);
+/// // Retrieve the subfont index for the requested glyph.
+/// let subfont_index = scaler.subfont_index(glyph_id);
+/// // Construct a subfont with the given configuration.
+/// let size = 16.0;
+/// let coords = &[];
+/// let with_hinting = false;
+/// let subfont = scaler.subfont(subfont_index, size, coords, with_hinting)?;
```

Comment:
When would the user use a subfont index other than the one computed for the glyph id? If the answer is never, why are we inflicting all this tiresome detail on them?

Reply:
If we really must inflict the details can we offer an interface that doesn't as well?

For context, I want people to be able to write code like https://github.com/rsheeter/read-fonts-wasm/blob/main/src/lib.rs and have it Just Work for truetype, cff, or cff2. Ideally I also want to interact with COLR similarly: if it says I need an outline for a glyph I can ask for it without having to care about the details of which outline table I'm using. At most I want to tell you something like what I consider the outline table priority to be, e.g. the order of tags.

Reply:
Once this lands, the code you linked will just work with CFF/CFF2 fonts :)